pkg/image/api/helper.go

```diff
@@ -642,7 +642,8 @@ func DifferentTagEvent(stream *ImageStream, tag string, next TagEvent) bool {
 	previous := &tags.Items[0]
 	sameRef := previous.DockerImageReference == next.DockerImageReference
 	sameImage := previous.Image == next.Image
-	return !(sameRef && sameImage)
+	sameGeneration := previous.Generation == next.Generation
+	return !(sameRef && sameImage && sameGeneration)
```

Comment:
I'll add the parameter and make sure to cover all, or at least most of the paths with new tests.